core/block.js

```diff
@@ -325,7 +325,8 @@ Blockly.Block.prototype.dispose = function(healStack) {
     this.workspace.removeChangeListener(this.onchangeWrapper_);
   }
 
-  if (Blockly.keyboardAccessibilityMode) {
+  if (Blockly.getMainWorkspace() &&
```

Comment:
The block has a workspace associated with it. why not use that?
This would affect blocks across workspaces if the mainWorkspace changes with a focus on the other workspace.

Reply:
fixed